web3/_utils/proof.py

```diff
@@ -0,0 +1,132 @@
+import rlp
+from rlp.sedes import (
+    Binary,
+    big_endian_int,
+)
+from web3._utils.encoding import (
+    pad_bytes,
+    pad_hex,
+)
+from eth_utils import (
+    keccak,
+)
+
+
+def storage_position(map_key_hex, position_hex):
+    '''
+    https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_getstorageat
+    Returns the patricia trie key (position) of a mapping's key.
+    This helper function is used for formating arguments to eth_getStorageAt
+    and eth_getProof.
+    @param map_key_hex is the key stored in a map
+    @param position_hex is the position where the mapping variable is defined
+        in the contract
+    '''
+    key = pad_hex(map_key_hex, 256)[2:] + pad_hex(position_hex, 256)[2:]
+    return keccak(bytes.fromhex(key)).hex()
+
+
+nibble_to_number = {}
+for i, c in enumerate('0123456789abcdef'):
+    nibble_to_number[c] = i
+
+
+class _Account(rlp.Serializable):
```

Comment:
Side note, it'd be nice if we could re-use the same data structure across `py-evm` and `web3.py`.  I believe that `eth-account` also has at least one copy of a serializable.  New library, something like:

- `eth-datatypes`

Reply:
I wasn't able to find something in eth-account that could be used instead of _Account(rlp.Serializable). Bwt i got this from [pyethereum](https://github.com/ethereum/pyethereum/blob/3c129aeaa374b223723e522fc2abb8baf6f3f142/ethereum/state.py#L53)